py/desiutil/names.py

```diff
@@ -43,14 +43,14 @@ def radec_to_desiname(target_ra, target_dec):
     target_ra, target_dec = np.atleast_1d(target_ra), np.atleast_1d(target_dec)
 
     inputs = {'target_ra': target_ra, 'target_dec': target_dec}
-    tests = {'NaN values': np.isnan,
-             'Infinite values': np.isinf,
-             'RA not in range [0, 360)': lambda x: (x < 0) | (x >= 360),
-             'Dec not in range [-90, 90]': lambda x: (x < -90) | (x > 90)}
+    tests = (('NaN values', np.isnan),
+             ('Infinite values', np.isinf),
+             ('RA not in range [0, 360)', lambda x: (x < 0) | (x >= 360)),
+             ('Dec not in range [-90, 90]', lambda x: (x < -90) | (x > 90)))
     for i in inputs:
-        for t in tests:
-            if (tests[t](inputs[i])).any():
-                raise ValueError(f"{t} detected in {i}!")
+        for key, check in tests:
+            if (check(inputs[i])).any():
+                raise ValueError(f"{key} detected in {i}!")
```

Comment:
This is very informative but could be made more informative with a `np.where()` call before we raise the error such that we can print the values within the array that are an issue.

I don't feel very strongly about this, but Stephen mentioned it in his comment to the issue so he may feel stronger.